module_3/gifts/controller/src/main/java/com/epam/esm/controller/view/PageView.java

```diff
@@ -0,0 +1,16 @@
+package com.epam.esm.controller.view;
+
+import java.util.List;
+
+import org.springframework.hateoas.RepresentationModel;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class PageView<T> extends RepresentationModel<PageView<T>> {
```

Comment:
This's a good way to solve issue, however you don't need to create a custom pageView with spring hateoas, because it contains a pageModel class
https://docs.spring.io/spring-hateoas/docs/current/reference/html/#fundamentals.representation-models